packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1891,17 +1853,17 @@ export class ImportResolver {
     }
 
     // Finds all of the stdlib modules and returns a Set containing all of their names.
-    private _buildStdlibCache(stdlibRoot: string | undefined): Set<string> {
+    private _buildStdlibCache(stdlibRoot: Uri | undefined): Set<string> {
         const cache = new Set<string>();
 
         if (stdlibRoot) {
-            const readDir = (root: string, prefix: string | undefined) => {
+            const readDir = (root: Uri, prefix: string | undefined) => {
                 this.readdirEntriesCached(root).forEach((entry) => {
                     if (entry.isDirectory()) {
-                        const dirRoot = combinePaths(root, entry.name);
+                        const dirRoot = root.combinePaths(entry.name);
                         readDir(dirRoot, prefix ? `${prefix}.${entry.name}` : entry.name);
                     } else if (entry.name.includes('.py')) {
-                        const stripped = stripFileExtension(entry.name);
+                        const stripped = entry.name.split('.py')[0];
```

Comment:
Should we still be calling `stripFileExtension` here? `split` is going to do the wrong thing for weird filenames like `weird.python.py`.

Reply:
Yes this looks like a mistake. I'll put it back.